rider-core/src/main/java/com/github/database/rider/core/dataset/DataSetExecutorImpl.java

```diff
@@ -861,7 +883,90 @@ public void compareCurrentDataSetWith(DataSetConfig expectedDataSetConfig, Strin
 
             DataSetAssertion.assertEqualsIgnoreCols(expectedTable, filteredActualTable, excludeCols);
         }
+    }
+
+    // TODO: refactor
+    private void compareProlog(IDataSet current, IDataSet expected, String[] tableNames) throws DatabaseUnitException {
+        StringBuilder sbFacts = new StringBuilder();
+        List<String> queryTerms = new ArrayList<>();
+
+        for (String tableName : tableNames) {
+            ITable expectedTable = null;
+            ITable actualTable = null;
+            try {
+                expectedTable = expected.getTable(tableName);
+                actualTable = current.getTable(tableName);
+            } catch (DataSetException e) {
+                throw new RuntimeException("DataSet comparison failed due to following exception: ", e);
+            }
 
+            for (int i = 0; i < actualTable.getRowCount(); i++) {
+                sbFacts.append(actualTable.getTableMetaData().getTableName().toLowerCase());
+                sbFacts.append("(");
+                ITable finalActualTable = actualTable;
+                int finalI = i;
+                sbFacts.append(
+                        Arrays.stream(actualTable.getTableMetaData().getColumns())
+                                .map(column -> {
+                                    try {
+                                        return finalActualTable.getValue(finalI, column.getColumnName());
+                                    } catch (DataSetException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                })
+                                .map(o -> "'" + o + "'")
+                                .collect(Collectors.joining(","))
+                );
+                sbFacts.append(").\n");
```

Comment:
Can we extract to something  like ` createActualTableFacts(sbFacts, actualTable, i);`?

I think it is clear if we return a String representing the tableFact instead of sbFacts as param, ex:

```
sbFacts.append(createTableFacts(actualTable, i));
```
Something like: 

```
private String createTableFacts(ITable actualTable, int i) throws DataSetException {
        StringBuilder tableFact = new StringBuilder();
        tableFact.append(actualTable.getTableMetaData().getTableName().toLowerCase());
        tableFact.append("(");
        ITable finalActualTable = actualTable;
        int finalI = i;
        tableFact.append(
                Arrays.stream(actualTable.getTableMetaData().getColumns())
                        .map(column -> {
                            try {
                                return finalActualTable.getValue(finalI, column.getColumnName());
                            } catch (DataSetException e) {
                                throw new RuntimeException(e);
                            }
                        })
                        .map(o -> "'" + o + "'")
                        .collect(Collectors.joining(","))
        );
        tableFact.append(").\n");
        
        return tableFact.toString();
    }
```